PublicFolders/src/SourceSideValidations/Get-IpmSubtree.ps1

```diff
@@ -6,57 +6,117 @@ function Get-IpmSubtree {
-                ForEach-Object {
-                    $progressCount++
-                    $currentFolder = $_.Identity.ToString()
-                    try {
-                        if ($sw.ElapsedMilliseconds -gt 1000) {
-                            $sw.Restart()
-                            Write-Progress @progressParams -Status $progressCount
-                        }
-
-                        [PSCustomObject]@{
-                            Identity          = $_.Identity.ToString()
-                            EntryId           = $_.EntryId.ToString()
-                            ParentEntryId     = $_.ParentFolder.ToString()
-                            DumpsterEntryId   = if ($_.DumpsterEntryId) { $_.DumpsterEntryId.ToString() } else { $null }
-                            FolderPathDepth   = $_.FolderPath.Depth
-                            FolderSize        = $_.FolderSize
-                            HasSubfolders     = $_.HasSubfolders
-                            ContentMailbox    = $_.ContentMailboxName
-                            MailEnabled       = $_.MailEnabled
-                            MailRecipientGuid = $_.MailRecipientGuid
-                            ItemCount         = 0
-                        }
-                    } catch {
-                        $errors++
-                        Write-Error -Message $currentFolder -Exception $_.Exception
-                        break
-                    }
+        $getCommand = { Get-PublicFolder -Recurse -ResultSize Unlimited }
+
+        if ($SlowTraversal) {
+            $getCommand = { $top = Get-PublicFolder "\"; Get-FoldersRecursive $top $foldersProcessed; $top }
+        }
+
+        $outputResultsScriptBlock = {
+            [CmdletBinding()]
+            param (
+                [Parameter(ValueFromPipeline = $true)]
+                [object]
+                $Folder
+            )
+
+            process {
+                $progressCount++
+
+                if ($sw.ElapsedMilliseconds -gt 1000) {
+                    $sw.Restart()
+                    Write-Progress @progressParams -Status $progressCount
+                }
+
+                $result = [PSCustomObject]@{
+                    Name              = $Folder.Name
+                    Identity          = $Folder.Identity.ToString()
+                    EntryId           = $Folder.EntryId.ToString()
+                    ParentEntryId     = $Folder.ParentFolder.ToString()
+                    DumpsterEntryId   = if ($Folder.DumpsterEntryId) { $Folder.DumpsterEntryId.ToString() } else { $null }
+                    FolderSize        = $Folder.FolderSize
+                    HasSubfolders     = $Folder.HasSubfolders
+                    ContentMailbox    = $Folder.ContentMailboxName
+                    MailEnabled       = $Folder.MailEnabled
+                    MailRecipientGuid = $Folder.MailRecipientGuid
+                }
+
+                $null = $ipmSubtree.Add($result)
```

Comment:
Why are we doing `$null = something`? Should is this the best way to do it vs a pipe to `Out-Null`?

Reply:
Actuall $null = is faster and [void] is even faster. Out-Null is the slowest approach:

https://docs.microsoft.com/en-us/powershell/scripting/dev-cross-plat/performance/script-authoring-considerations

I'm gradually migrating to [void] even though I don't like how it looks. I went ahead and updated it in this location.